app/services/AuthenticationService.scala

```diff
@@ -1,8 +1,87 @@
 package services
 
-import com.gu.identity.cookie.IdentityKeys
-import configuration.Config
+import java.util.concurrent.Executors
 
-object AuthenticationService extends com.gu.identity.play.AuthenticationService {
-  override val identityKeys: IdentityKeys = Config.Identity.keys
+import com.gu.identity.auth.UserCredentials
+import com.gu.identity.model.User
+import com.gu.identity.play.IdentityPlayAuthService
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger._
+import org.http4s.Uri
+import play.api.mvc.{Cookie, RequestHeader}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+// The following classes were previously defined in identity-play-auth,
+// but have been removed as part of the changes to that library:
+// authenticating users via a call to identity API; removing periphery functionality.
+// They have been redefined here to reduce diff across PRs,
+// but these classes could get refactored / simplified / removed in subsequent PRs.
+sealed trait AccessCredentials
+object AccessCredentials {
+  case class Cookies(scGuU: String, guU: Option[String] = None) extends AccessCredentials {
+    val cookies: Seq[Cookie] = Seq(
+      Cookie(name = "SC_GU_U", scGuU)
+    ) ++ guU.map(c => Cookie(name = "GU_U", c))
+  }
+  case class Token(tokenText: String) extends AccessCredentials
+}
+case class IdMinimalUser(id: String, displayName: Option[String])
+case class AuthenticatedIdUser(credentials: AccessCredentials, user: IdMinimalUser)
+
+// Class used to authenticate a user. Authentication is done by calling back to identity API.
+// This is so session invalidation can be taken into account when authenticating a user.
+// See comments on IdentityPlayAuthService for more information.
+class AuthenticationService private (identityPlayAuthService: IdentityPlayAuthService) {
+
+  import AuthenticationService._
+
+  // Even though a request to identity API is made when this method is called,
+  // the IO instance returned by getUserFromRequest() is executed synchronously,
+  // as changing the result type to Future resulted in a PR with a very complicated diff:
+  // https://github.com/guardian/subscriptions-frontend/pull/1265
+  //
+  // It was deemed that the risk of a bug being introduced in #1265
+  // was greater than the risk of a user getting a slower than normal response
+  // (as a result of all threads available to execute this method call being blocked).
+  // The latter was considered low risk, since the subscriptions-frontend app is low traffic
+  // and a dedicated thread pool of 30 threads has been reserved
+  // to execute these blocking calls (c.f. Authentication.unsafeInit()).
+  // The fact that subscriptions-frontend is a deprecated application also factored into the decision.
```

Comment:
sounds very fair, even though it offends my functional purity mind 🙈 
given how limited future development this app will get, the argument in favour of maintainability is reduced a lot!